### PR TITLE
2021 32x hacks

### PR DESCRIPTION
**Pull request by blinard-BIOINFO**
This PR does the following:

Spotted by @hotelvictorcharlie , in #720 :
- corrects some rom name spelling mistakes
- corrects 2 entries : checksum of Doom 32X Delta Edition Episode 1 and 3 (episode 2 unchanged). They now all match the most recent romhacking.net patch and not the older version from doomworld. 

Doom 32X Hacks updates :
- adds 4 new 'Doom 32X hacks' entries, all patches come from romhacking.net :  
  - Doom 32X Delta Edition episode 4 
  - Doom 32X Delta Edition episode 5
  - European region patch variant for 'Ultimate Doom All Episodes'
  - European region patch variant for 'Doom 2 All Levels Hack Maps'
- updates Doom 32X Resurrection from hack v1.3 to v1.4 (splitscreen coop added !!!)

### CODE REVIEW

File: EverDrive Pack SMDBs/Mega EverDrive SMDB.txt [hunk 1]
```diff
@@ -1837,18 +1837,22 @@ a68fef8167be7be2d4a15f0711f49e5de30c1cd348bc51334e67905774f42229	Mega EverDrive/
...
+860d2485d8b9d18a3d2c3a54ea321c98fab0603faa130d4dd96b6dbb3e610d48	Mega EverDrive/3 Sega 32X/4 Homebrew & Hacks/Doom 32X Hacks/Doom 2 All Levels Hack Maps (Japan, USA) (Hack) v1.03 e-Doctor.32x	f23ead65bb6dac3f6481ee44fead2a684035d5ef	9c1935bc2f865778b03ca85545d7ce92	e0d57485
+7892df5a8e0fd83cd5b9ce3d85921a3a4a432bd02fa80d30c21c4d5bf1281a08	Mega EverDrive/3 Sega 32X/4 Homebrew & Hacks/Doom 32X Hacks/Doom 2 All Levels Hack Maps (Europe) (Hack) v1.03 e-Doctor.32x	0fe46fc4c1264f01d1202caee1eb669cdb04f6c9	9d0cd4a76312b409e188a32c482c7183	5f251d3c
+312b203a86f67ccba3f981c3d62f4813e9051c2cf0799f2644bcc79e3f1a6586	Mega EverDrive/3 Sega 32X/4 Homebrew & Hacks/Doom 32X Hacks/Doom 32X - 24 Levels expansion + Addendum (Europe) (Hack) v1.1 Saxman & Nuxius.32x	0de2eaccae8468acbea764140cc8f76474dda237	7c2144bc01cb16b2c12f0c3913e9d3cb	4821de9c
+f1deadad3f7ad6115dfb35147cf1b98dca29c0073e4f6c29cf11be4559782b2b	Mega EverDrive/3 Sega 32X/4 Homebrew & Hacks/Doom 32X Hacks/Doom 32X - Delta Edition - Episode 1 (Japan, USA) (Hack) v1.0 Roeblez.32x	248af9e48314dea3ed401da248b415ddfa1e96a6	53c0281b98d5bbc695e0d736585d7494	d2471a0a
```
Comment (hotelvictorcharlie) [inline EverDrive Pack SMDBs/Mega EverDrive SMDB.txt:1843]: Should be spelled [Roebloz](https://www.doomworld.com/profile/30346-roebloz/), not Roeblez.

**Review by hotelvictorcharlie — CHANGES_REQUESTED** [review]

File: EverDrive Pack SMDBs/Mega EverDrive SMDB.txt [hunk 2]
```diff
@@ -1843,7 +1843,7 @@ a68fef8167be7be2d4a15f0711f49e5de30c1cd348bc51334e67905774f42229	Mega EverDrive/
...
 330546cafb9c8ca6aee917f64ad767ecb220176bff3f21e4d1ac3237f4ac962a	Mega EverDrive/3 Sega 32X/4 Homebrew & Hacks/Doom 32X Hacks/Doom 32X - Delta Edition - Episode 2 (Japan, USA) (Hack) v1.0 Roeblez.32x	8897cbf291193a33b59621d7b66b8d94a471988f	6af06cb82b84b46bca3ca4e4e065c16a	157caeab
 468a823576353ea3e5e1c0812811356bd02d296d4ce11edb5fc13e6d409a1581	Mega EverDrive/3 Sega 32X/4 Homebrew & Hacks/Doom 32X Hacks/Doom 32X - Delta Edition - Episode 3 (Japan, USA) (Hack) v1.0 Roeblez.32x	5429b0fca75d3925077a53957fb3696e613c05ea	b588ba6326357aa36da5353cac91eb1a	8e1ac721
-2a61fb0f98f4336f65978bd745fb8775b9aa6ec71ec6342aafb2b6edee85fc3b	Mega EverDrive/3 Sega 32X/4 Homebrew & Hacks/Doom 32X Hacks/Doom 32X Resurrection (Europe, Japan, USA) (Hack) v1.3 D32XR Team.32x	22850ad1a75a3c74a11da4dd95307aeaa7963b85	b8aec735bcd311a1380f37564535e30b	bd79e2f0
+40fc27d6e6cef3434902e17ac518b9ed932ff046b947477a9b158d5c6091b775	Mega EverDrive/3 Sega 32X/4 Homebrew & Hacks/Doom 32X Hacks/Doom 32X Resurrection (Europe, Japan, USA) (Hack) v1.4 D32XR Team.32x	9a7b333899eefe57a975c318477ceb5de91c0386	d2625fec17662ad2c1cc1829ac9e4180	d2000095
```
Comment (frederic-mahe) [inline EverDrive Pack SMDBs/Mega EverDrive SMDB.txt:1846]: https://github.com/frederic-mahe/Hardware-Target-Game-Database/pull/721#issuecomment-995991739

 @Infinitum8t says that there is now a v1.5 of _Doom 32X Resurrection_.

**Comment by blinard-BIOINFO** [comment]
Indeed a hack update has been released a few days just after this new PR.  XD
I push the update this week-end.

File: EverDrive Pack SMDBs/Mega EverDrive SMDB.txt (hunk 1 above)
Comment (blinard-BIOINFO) [reply EverDrive Pack SMDBs/Mega EverDrive SMDB.txt:1843]: Thanks. I will push the fix this week-end.

File: EverDrive Pack SMDBs/Mega EverDrive SMDB.txt (hunk 2 above)
Comment (blinard-BIOINFO) [reply EverDrive Pack SMDBs/Mega EverDrive SMDB.txt:1846]: Indeed a hack update has been released a few days just after this new PR. XD
I push the update this week-end.

**Comment by frederic-mahe** [comment]
Thanks @blinard-BIOINFO sorry you have to go through so many hoops.

**Comment by blinard-BIOINFO** [comment]
> so many hoops
No worries, I'm happy to do these minor contributions to this huge database.

**Comment by blinard-BIOINFO** [comment]
@frederic-mahe 
the last commit answers @hotelvictorcharlie + @infinitum8t comments

**Review by hotelvictorcharlie — APPROVED** [review]

**Pull request merged**